src/test/java/org/sasanlabs/service/vulnerability/sqlInjection/UnionBasedSQLInjectionVulnerabilityTest.java

```diff
@@ -0,0 +1,87 @@
+package org.sasanlabs.service.vulnerability.sqlInjection;
+
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.PreparedStatementSetter;
+import org.springframework.jdbc.core.ResultSetExtractor;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.*;
+
+class UnionBasedSQLInjectionVulnerabilityTest {
+
+    private UnionBasedSQLInjectionVulnerability unionBasedSQLInjectionVulnerability;
+    private JdbcTemplate template;
+
+    @BeforeEach
+    void setUp() throws IOException {
+        template = Mockito.spy(new JdbcTemplate());
+
+        doReturn(null)
+                .when(template)
+                .query(anyString(), (ResultSetExtractor<? extends Object>) any());
+
+        unionBasedSQLInjectionVulnerability = Mockito.spy(new UnionBasedSQLInjectionVulnerability(template));
+    }
+
+    @Test
+    void getCarInformationLevel1_ExpectParamInjected() throws IOException {
+        // Act
+        final Map<String, String> params = new HashMap<>();
+        params.put("id", "1 UNION SELECT * FROM cars;");
+        unionBasedSQLInjectionVulnerability.getCarInformationLevel1(params);
+
+        // Assert
+        verify(template).query(eq("select * from cars where id=1 UNION SELECT * FROM cars;"), (ResultSetExtractor<? extends Object>) any());
+    }
+
+    @Test
+    void getCarInformationLevel2_ExpectParamInjected() throws IOException {
+        // Act
+        final Map<String, String> params = new HashMap<>();
+        params.put("id", "1' UNION SELECT * FROM cars; --");
+        unionBasedSQLInjectionVulnerability.getCarInformationLevel2(params);
+
+        // Assert
+        verify(template).query(eq("select * from cars where id='1' UNION SELECT * FROM cars; --'"), (ResultSetExtractor<? extends Object>) any());
+    }
+
+    @Test
+    void getCarInformationLevel3_ExpectParamEscaped() throws IOException {
+        // Act
+        final Map<String, String> params = new HashMap<>();
+        params.put("id", "1' UNION SELECT * FROM cars; --");
+        unionBasedSQLInjectionVulnerability.getCarInformationLevel3(params);
+
+        // Assert
+        verify(template).query(eq("select * from cars where id='1 UNION SELECT * FROM cars; --'"), (ResultSetExtractor<? extends Object>) any());
+
+    }
+
+    @Test
+    void getCarInformationLevel4_ExpecParamEscaped() throws IOException {
+        // Setup
+        template = Mockito.spy(new JdbcTemplate());
+        doReturn(null)
```

Comment:
I think you can directly define the mock on same template method. can you please check it once.